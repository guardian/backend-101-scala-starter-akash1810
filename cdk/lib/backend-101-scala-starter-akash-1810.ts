import { GuPlayApp } from '@guardian/cdk';
import { AccessScope } from '@guardian/cdk/lib/constants';
import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import {
	GuDistributionBucketParameter,
	GuStack,
} from '@guardian/cdk/lib/constructs/core';
import { GuCname } from '@guardian/cdk/lib/constructs/dns';
import type { App } from 'aws-cdk-lib';
import { Duration } from 'aws-cdk-lib';
import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';

export class Backend101ScalaStarterAkash1810 extends GuStack {
	constructor(scope: App, id: string, props: Omit<GuStackProps, 'env'>) {
		super(scope, id, {
			...props,
			env: { region: 'eu-west-1' },
		});

		const app = 'backend-101-scala-starter-akash1810';
		const domainName = `backend101-akash1810.gutools.co.uk`;

		const s3Path = [
			GuDistributionBucketParameter.getInstance(this).valueAsString,
			this.stack,
			this.stage,
			app,
		].join('/');

		const { loadBalancer } = new GuPlayApp(this, {
			app,
			access: { scope: AccessScope.PUBLIC },
			certificateProps: {
				domainName,
			},
			instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
			monitoringConfiguration: { noMonitoring: true },
			scaling: { minimumInstances: 1 },
			imageRecipe: 'backend-101',
			userData: [
				'#!/bin/bash -ev',
				`aws s3 cp s3://${s3Path}/app.service /etc/systemd/system/${app}.service`,
				`aws s3 cp s3://${s3Path}/hello-world.jar /hello-world.jar`,
				`systemctl start ${app}`,
			].join('\n'),
		});

		new GuCname(this, 'dns-record', {
			app,
			domainName,
			ttl: Duration.minutes(1),
			resourceRecord: loadBalancer.loadBalancerDnsName,
		});
	}
}
