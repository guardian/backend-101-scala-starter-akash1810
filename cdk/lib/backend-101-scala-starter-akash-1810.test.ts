import { GuRoot } from '@guardian/cdk/lib/constructs/root';
import { Template } from 'aws-cdk-lib/assertions';
import { Backend101ScalaStarterAkash1810 } from './backend-101-scala-starter-akash-1810';

describe('The Backend101ScalaStarterAkash1810 stack', () => {
	it('matches the snapshot', () => {
		const app = new GuRoot();
		const stack = new Backend101ScalaStarterAkash1810(
			app,
			'Backend101ScalaStarterAkash1810',
			{ stack: 'deploy', stage: 'TEST' },
		);
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
