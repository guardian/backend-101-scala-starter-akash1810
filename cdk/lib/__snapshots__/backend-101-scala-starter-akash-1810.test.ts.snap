// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Backend101ScalaStarterAkash1810 stack matches the snapshot 1`] = `
Object {
  "Outputs": Object {
    "LoadBalancerBackend101scalastarterakash1810DnsName": Object {
      "Description": "DNS entry for LoadBalancerBackend101scalastarterakash1810",
      "Value": Object {
        "Fn::GetAtt": Array [
          "LoadBalancerBackend101scalastarterakash1810C36549AB",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": Object {
    "AMIBackend101scalastarterakash1810": Object {
      "Description": "Amazon Machine Image ID for the app backend-101-scala-starter-akash1810. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "DistributionBucketName": Object {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "LoggingStreamName": Object {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "VpcId": Object {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "backend101scalastarterakash1810PrivateSubnets": Object {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "backend101scalastarterakash1810PublicSubnets": Object {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": Object {
    "AutoScalingGroupBackend101scalastarterakash1810ASG0ABA0C09": Object {
      "Properties": Object {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchConfigurationName": Object {
          "Ref": "AutoScalingGroupBackend101scalastarterakash1810LaunchConfig4BD0D963",
        },
        "MaxSize": "2",
        "MinSize": "1",
        "Tags": Array [
          Object {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:pattern-name",
            "PropagateAtLaunch": true,
            "Value": "GuPlayApp",
          },
          Object {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "LogKinesisStreamName",
            "PropagateAtLaunch": true,
            "Value": Object {
              "Ref": "LoggingStreamName",
            },
          },
          Object {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "Backend101ScalaStarterAkash1810/AutoScalingGroupBackend101scalastarterakash1810",
          },
          Object {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
        ],
        "TargetGroupARNs": Array [
          Object {
            "Ref": "TargetGroupBackend101scalastarterakash18108F7457DD",
          },
        ],
        "VPCZoneIdentifier": Object {
          "Ref": "backend101scalastarterakash1810PrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "AutoScalingGroupBackend101scalastarterakash1810InstanceProfile1F3AA155": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleBackend101scalastarterakash1810687FA801",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "AutoScalingGroupBackend101scalastarterakash1810LaunchConfig4BD0D963": Object {
      "DependsOn": Array [
        "InstanceRoleBackend101scalastarterakash1810687FA801",
      ],
      "Properties": Object {
        "IamInstanceProfile": Object {
          "Ref": "AutoScalingGroupBackend101scalastarterakash1810InstanceProfile1F3AA155",
        },
        "ImageId": Object {
          "Ref": "AMIBackend101scalastarterakash1810",
        },
        "InstanceType": "t4g.small",
        "MetadataOptions": Object {
          "HttpTokens": "required",
        },
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "GuHttpsEgressSecurityGroupBackend101scalastarterakash181073587ECE",
              "GroupId",
            ],
          },
          Object {
            "Fn::GetAtt": Array [
              "WazuhSecurityGroup",
              "GroupId",
            ],
          },
        ],
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash -ev
aws s3 cp s3://",
                Object {
                  "Ref": "DistributionBucketName",
                },
                "/deploy/TEST/backend-101-scala-starter-akash1810/app.service /etc/systemd/system/backend-101-scala-starter-akash1810.service
aws s3 cp s3://",
                Object {
                  "Ref": "DistributionBucketName",
                },
                "/deploy/TEST/backend-101-scala-starter-akash1810/hello-world.jar /hello-world.jar
systemctl start backend-101-scala-starter-akash1810",
              ],
            ],
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "CertificateBackend101scalastarterakash18108F3BCFBD": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "DomainName": "backend101-akash1810.gutools.co.uk",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "DescribeEC2PolicyFF5F9295": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleBackend101scalastarterakash1810687FA801",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GetDistributablePolicyBackend101scalastarterakash1810518EE323": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:s3:::",
                    Object {
                      "Ref": "DistributionBucketName",
                    },
                    "/deploy/TEST/backend-101-scala-starter-akash1810/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyBackend101scalastarterakash1810518EE323",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleBackend101scalastarterakash1810687FA801",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupBackend101scalastarterakash181073587ECE": Object {
      "Properties": Object {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupBackend101scalastarterakash1810fromBackend101ScalaStarterAkash1810LoadBalancerBackend101scalastarterakash1810SecurityGroup2F9D72929000055D54EC": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 9000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupBackend101scalastarterakash181073587ECE",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "LoadBalancerBackend101scalastarterakash1810SecurityGroupD2E74D38",
            "GroupId",
          ],
        },
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuLogShippingPolicy981BFE5A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:kinesis:eu-west-1:",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    Object {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleBackend101scalastarterakash1810687FA801",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InstanceRoleBackend101scalastarterakash1810687FA801": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ListenerBackend101scalastarterakash181087DBE05A": Object {
      "Properties": Object {
        "Certificates": Array [
          Object {
            "CertificateArn": Object {
              "Ref": "CertificateBackend101scalastarterakash18108F3BCFBD",
            },
          },
        ],
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "TargetGroupBackend101scalastarterakash18108F7457DD",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "LoadBalancerBackend101scalastarterakash1810C36549AB",
        },
        "Port": 443,
        "Protocol": "HTTPS",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerBackend101scalastarterakash1810C36549AB": Object {
      "Properties": Object {
        "LoadBalancerAttributes": Array [
          Object {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "LoadBalancerBackend101scalastarterakash1810SecurityGroupD2E74D38",
              "GroupId",
            ],
          },
        ],
        "Subnets": Object {
          "Ref": "backend101scalastarterakash1810PublicSubnets",
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerBackend101scalastarterakash1810SecurityGroupD2E74D38": Object {
      "Properties": Object {
        "GroupDescription": "Automatically created Security Group for ELB Backend101ScalaStarterAkash1810LoadBalancerBackend101scalastarterakash18108592EF48",
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerBackend101scalastarterakash1810SecurityGrouptoBackend101ScalaStarterAkash1810GuHttpsEgressSecurityGroupBackend101scalastarterakash181015ABEE7B900063273FBA": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupBackend101scalastarterakash181073587ECE",
            "GroupId",
          ],
        },
        "FromPort": 9000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "LoadBalancerBackend101scalastarterakash1810SecurityGroupD2E74D38",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ParameterStoreReadBackend101scalastarterakash1810F3F76E22": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:eu-west-1:",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/deploy/backend-101-scala-starter-akash1810",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:eu-west-1:",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/deploy/backend-101-scala-starter-akash1810/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleBackend101scalastarterakash1810687FA801",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SSMRunCommandPolicy244E1613": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
                "ssm:UpdateInstanceInformation",
                "ssm:ListInstanceAssociations",
                "ssm:DescribeInstanceProperties",
                "ssm:DescribeDocumentParameters",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ssm-run-command-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleBackend101scalastarterakash1810687FA801",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupBackend101scalastarterakash18108F7457DD": Object {
      "Properties": Object {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 5,
        "Port": 9000,
        "Protocol": "HTTP",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TargetGroupAttributes": Array [
          Object {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
          Object {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/backend-101-scala-starter-akash1810",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "dnsrecord": Object {
      "Properties": Object {
        "Name": "backend101-akash1810.gutools.co.uk",
        "RecordType": "CNAME",
        "ResourceRecords": Array [
          Object {
            "Fn::GetAtt": Array [
              "LoadBalancerBackend101scalastarterakash1810C36549AB",
              "DNSName",
            ],
          },
        ],
        "Stage": "TEST",
        "TTL": 60,
      },
      "Type": "Guardian::DNS::RecordSet",
    },
  },
}
`;
